src/main/java/appeng/parts/automation/ItemPickupStrategy.java

```diff
@@ -114,29 +114,18 @@ public Result tryStartPickup(IEnergySource energySource, PickupSink sink) {
         return Result.CANT_PICKUP;
     }
 
-    @Override
-    public void completePickup(IEnergySource energySource, PickupSink sink) {
-
-        var blockState = level.getBlockState(pos);
-        if (!this.canHandleBlock(level, pos, blockState)) {
-            return;
-        }
-
-        var items = this.obtainBlockDrops(level, pos);
-        var requiredPower = this.calculateEnergyUsage(level, pos, items);
-
+    private void completePickup(IEnergySource energySource, PickupSink sink, List<ItemStack> items, float requiredPower,
+            BlockState blockState) {
         if (!this.breakBlockAndStoreExtraItems(sink, level, pos)) {
             // We failed to actually replace the block with air, or it already was the case
             return;
         }
 
         for (var item : items) {
             var inserted = storeItemStack(sink, item);
-            // If inserting the item fully was not possible, drop it as an item entity instead if the storage clears up,
-            // we'll pick it up that way
             if (inserted < item.getCount()) {
-                item.shrink(inserted);
-                Platform.spawnDrops(level, pos, Collections.singletonList(item));
+                // Shouldn't happen unless something messes with state from another thread?
+                throw new IllegalStateException("Failed to insert items after simulated insert succeeded");
```

Comment:
Never trust simulation! This should not be changed.

Try picking up a chest that's connected to a storage bus. By the time we get here the chest is gone and the storage bus will reject the insertion. (I think)